ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -162,7 +160,7 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
 
   @Override
   public void processRequestCharges(RestRequest restRequest, RestResponseChannel responseChannel, BlobInfo blobInfo) {
-    Map<QuotaName, Double> requestCost = requestCostPolicy.calculateRequestCost(restRequest, responseChannel, blobInfo)
+    Map<QuotaName, Double> requestCost = ambryCostModelPolicy.calculateSCERequestCost(restRequest, responseChannel, blobInfo)
```

Comment:
I assume that at the end of the day, we are going to remove the quota logic in this method and AmbryCostModelPolicy?

Reply:
So Ambry's cost calculation for quota purposes, and for cost-model have diverged. We are using different policies to calculate these costs. Thats the premise of this PR -- to separate out these two policies so that they can be extended independently in future, if any extension required.

There are two places in `AmbrySecurityService` where quota check is done currently. One in `postProcessRequest` and the other in `processResponse`(`processResquestCharges`). We can definitely get rid of quota check in `postProcessRequest`.
However the calculations in `processResquestCharges` are needed at the end of a user request, to send back the quota header responses as part of users' request response. Same applies to cost model policy calculations to send back cost headers as part of response. So the quota logic in `processRequestCharges` method should remain in my opinion.

There's an orthogonal question of whether `AmbryCostModelPolicy` needs the concept of `QuotaName` as part of the type of costs it handles. I have tried to remove any reference to `QuotaName` from the general interfaces. At the present moment though, the cost model policy implementation (`SimpleAmbryCostModelPolicy`) does use `QuotaName` and `QuotaUtils`. We can try to separate that out later, or maybe rename `QuotaName` enum such that it can be more generic.

Reply:
I agree, the `AmbryCostModelPolicy` shouldn't have quota name since it has nothing to do with quota now.